gbus/bus.go

```diff
@@ -549,17 +551,23 @@ func (b *DefaultBus) monitorAMQPErrors() {
 
 func (b *DefaultBus) sendImpl(ctx context.Context, tx *sql.Tx, toService, replyTo, exchange, topic string, message *BusMessage, policies ...MessagePolicy) (er error) {
 	b.SenderLock.Lock()
-	defer b.SenderLock.Unlock()
-
+	span, ctx := opentracing.StartSpanFromContext(ctx, "sendImpl")
 	defer func() {
 		if err := recover(); err != nil {
 
 			errMsg := fmt.Sprintf("panic recovered panicking err:\n%v\n%s", err, debug.Stack())
 			er = errors.New(errMsg)
+			span.LogFields(slog.Error(er))
 		}
+		b.SenderLock.Unlock()
```

Comment:
why do you need the error block to be included in the synced section?

Reply:
It's not an error block, it's part of the defer.

Reply:
yes, that is what I meant, why split the b.SenderLock.Lock() and defer b.SenderLock.Unlock() to separate code bloacks ?